src/extractor/suffix_table.cpp

```diff
@@ -0,0 +1,48 @@
+#include "extractor/suffix_table.hpp"
+
+#include "util/lua_util.hpp"
+#include "util/simple_logger.hpp"
+
+#include <boost/assert.hpp>
+#include <boost/ref.hpp>
+#include <boost/algorithm/string.hpp>
+
+#include <vector>
+
+namespace osrm
+{
+namespace extractor
+{
+
+void SuffixTable::fill(lua_State *lua_state)
+{
+    BOOST_ASSERT(lua_state != nullptr);
+    if (!util::luaFunctionExists(lua_state, "get_name_suffix_list"))
+        return;
+
+    std::vector<std::string> suffixes_vector;
+    try
+    {
+        // call lua profile to compute turn penalty
+        luabind::call_function<void>(lua_state, "get_name_suffix_list",
+                                     boost::ref(suffixes_vector));
+    }
+    catch (const luabind::error &er)
+    {
+        util::SimpleLogger().Write(logWARNING) << er.what();
+    }
+
+    for (auto suffix : suffixes_vector)
+    {
+        boost::algorithm::to_lower(suffix);
+        suffix_set.insert(suffix);
```

Comment:
I would do a bulk to_lower + insert instead if working on one item after the other.

    auto suffix_view = suffixes_vector | transformed(to_lower);
    set.insert(begin(suffixes_view), end(suffixes_view);

or similar --- this still only does a single pass over the vector and the set knows the size in advance